api/proto/teleport/userprovisioning/v2/statichostuser.proto

```diff
@@ -29,30 +29,33 @@ message StaticHostUser {
   // sub_kind is an optional resource sub kind, used in some resources.
   string sub_kind = 2;
   // version is the resource version. It must be specified.
-  // Supported values are: `v1`.
+  // Supported values are: `v2`.
   string version = 3;
   // metadata is resource metadata.
   teleport.header.v1.Metadata metadata = 4;
   // spec is the static host user spec.
   StaticHostUserSpec spec = 5;
 }
 
-// StaticHostUserSpec is the static host user spec.
-message StaticHostUserSpec {
-  // login is the login to create on the node.
-  string login = 1;
-  // groups is a list of additional groups to add the user to.
-  repeated string groups = 2;
-  // sudoers is a list of sudoer entries to add.
-  repeated string sudoers = 3;
-  // uid is the new user's uid.
-  string uid = 4;
-  // gid is the new user's gid.
-  string gid = 5;
+// Matcher is a matcher for nodes to add the user to.
+message Matcher {
   // node_labels is a map of node labels that will create a user from this
   // resource.
-  wrappers.LabelValues node_labels = 6;
+  repeated teleport.label.v1.Label node_labels = 1;
   // node_labels_expression is a predicate expression to create a user from
   // this resource.
-  string node_labels_expression = 7;
+  string node_labels_expression = 2;
+  // groups is a list of additional groups to add the user to.
+  repeated string groups = 3;
+  // sudoers is a list of sudoer entries to add.
+  repeated string sudoers = 4;
+  // uid is the new user's uid.
+  int32 uid = 5;
+  // gid is the new user's gid.
+  int32 gid = 6;
```

Comment:
sort of nit: other than `fixed32` and `fixed64`, protobuf numbers are variable length based on the value they contain, so there's no space saving by using `int32` over `int64`, and nothing in POSIX prevents uid_t from being 64 bits - and seeing as we're likely going to have to cast the values to a Go `int` anyway:
```suggestion
  // uid is the new user's uid.
  int64 uid = 5;
  // gid is the new user's gid.
  int64 gid = 6;
```